### PR TITLE
Made change for filtering top consumer based on the ceph storage class and handling the Inventory

### PR DESCRIPTION
**Pull request by cloudbehl**
Signed-off-by: Ankush Behl <cloudbehl@gmail.com>

### CODE REVIEW

**Comment by cloudbehl** [comment]
@rawagner @gnehapk @afreen23 can you review this PR.

File: frontend/packages/ceph-storage-plugin/src/selectors/index.ts [hunk 1]
```diff
@@ -24,3 +27,6 @@ export const getCephPVs = (pvsData: K8sResourceKind[] = []): K8sResourceKind[] =
 
 export const getCephNodes = (nodesData: K8sResourceKind[] = []): K8sResourceKind[] =>
   nodesData.filter((node) => _.keys(_.get(node, 'metadata.labels')).includes(cephStorageLabel));
+
+export const getCephSC = (scData: K8sResourceKind[]): K8sResourceKind[] =>
+  scData.filter((sc) => cephStorageProvisioners.includes(_.get(sc, 'provisioner')));
```
Comment (gnehapk) [inline frontend/packages/ceph-storage-plugin/src/selectors/index.ts]: Nit: No need to have brackets around `sc` parameter.

File: frontend/packages/ceph-storage-plugin/src/selectors/index.ts [hunk 2]
```diff
@@ -2,18 +2,21 @@ import * as _ from 'lodash';
...
-  pvcsData.filter((pvc) =>
-    cephStorageProvisioners.includes(
-      _.get(pvc, 'metadata.annotations["volume.beta.kubernetes.io/storage-provisioner"]'),
-    ),
-  );
+export const getCephPVCs = (
+  cephScNames: string[] = [],
+  pvcsData: K8sResourceKind[] = [],
+): K8sResourceKind[] =>
+  pvcsData.filter((pvc) => cephScNames.includes(_.get(pvc, 'spec.storageClassName')));
```
Comment (gnehapk) [inline frontend/packages/ceph-storage-plugin/src/selectors/index.ts]: Nit: No need to have brackets around `pvc` parameter. Please check this across rest of the page.

Comment (cloudbehl) [reply frontend/packages/ceph-storage-plugin/src/selectors/index.ts]: @gnehapk automatically added by linter.

File: frontend/packages/ceph-storage-plugin/src/selectors/index.ts (hunk 1 above)
Comment (cloudbehl) [reply frontend/packages/ceph-storage-plugin/src/selectors/index.ts]: @gnehapk automatically added by linter.

File: frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/inventory-card.tsx [hunk 3]
```diff
@@ -65,6 +71,10 @@ const InventoryCard: React.FC<DashboardItemProps> = ({
   const pvsLoadError = _.get(resources.pvs, 'loadError');
   const pvsData = _.get(resources.pvs, 'data', []) as K8sResourceKind[];
 
+  const scData = _.get(resources.sc, 'data', []) as K8sResourceKind[];
+  const filteredCephSc = getCephSC(scData);
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/inventory-card.tsx]:
```suggestion
  const filteredCephSC = getCephSC(scData);
```
Its a nit but I think we should keep the SC naming convention consistent throughout.

File: frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/inventory-card.tsx [hunk 4]
```diff
@@ -65,6 +71,10 @@ const InventoryCard: React.FC<DashboardItemProps> = ({
...
+  const filteredSCNames = filteredCephSc.map((sc) => _.get(sc, 'metadata.name'));
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/inventory-card.tsx]:
```suggestion
  const filteredSCNames = filteredCephSC.map((sc) => _.get(sc, 'metadata.name'));
```
Same thing here

File: frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/top-consumers-card/top-consumers-card-body.tsx [hunk 5]
```diff
@@ -30,32 +30,42 @@ const chartLegendPropsValue = {
...
-  Math.floor((6 * value) / 4),
+  _.round(value / 4, 1),
+  _.round(value / 2, 1),
+  _.round((3 * value) / 4, 1),
+  _.round(value, 1),
+  _.round((5 * value) / 4, 1),
+  _.round((6 * value) / 4, 1),
 ];
 
+const sortResources: SortResourcesProps = (a, b) => {
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/top-consumers-card/top-consumers-card-body.tsx]: Feels like this should go to utils file and we should import it there.

File: frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/top-consumers-card/top-consumers-card-body.tsx [hunk 6]
```diff
@@ -30,32 +30,42 @@ const chartLegendPropsValue = {
 };
 
 const getYTickValues = (value: number): number[] => [
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/top-consumers-card/top-consumers-card-body.tsx]: Feels like this should go to utils file and we should import it from there.

File: frontend/packages/ceph-storage-plugin/src/selectors/index.ts [hunk 7]
```diff
@@ -2,18 +2,21 @@ import * as _ from 'lodash';
...
   alerts.filter((alert) => _.get(alert, 'annotations.storage_type') === 'ceph');
 
-export const getCephPVCs = (pvcsData: K8sResourceKind[] = []): K8sResourceKind[] =>
-  pvcsData.filter((pvc) =>
-    cephStorageProvisioners.includes(
-      _.get(pvc, 'metadata.annotations["volume.beta.kubernetes.io/storage-provisioner"]'),
-    ),
-  );
+export const getCephPVCs = (
+  cephScNames: string[] = [],
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/src/selectors/index.ts]:
```suggestion
  cephSCNames: string[] = [],
```
SC would be more descriptive.

File: frontend/packages/ceph-storage-plugin/src/selectors/index.ts (hunk 2 above)
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/src/selectors/index.ts]:
```suggestion
  pvcsData.filter((pvc) => cephSCNames.includes(_.get(pvc, 'spec.storageClassName')));
```
Similarly here too.

File: frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/inventory-card.tsx [hunk 8]
```diff
@@ -65,6 +71,10 @@ const InventoryCard: React.FC<DashboardItemProps> = ({
   const pvsLoadError = _.get(resources.pvs, 'loadError');
   const pvsData = _.get(resources.pvs, 'data', []) as K8sResourceKind[];
 
+  const scData = _.get(resources.sc, 'data', []) as K8sResourceKind[];
+  const filteredCephSC = getCephSC(scData);
+  const filteredSCNames = filteredCephSC.map((sc) => _.get(sc, 'metadata.name'));
```
Comment (rawagner) [inline frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/inventory-card.tsx]: nit: you can use `getName` selector from `console-shared`

File: frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/top-consumers-card/top-consumers-card-body.tsx [hunk 9]
```diff
@@ -29,14 +29,11 @@ const chartLegendPropsValue = {
...
-  Math.floor(value / 2),
-  Math.floor((3 * value) / 4),
-  Math.floor(value),
-  Math.floor((5 * value) / 4),
-  Math.floor((6 * value) / 4),
-];
+const getMaxCapacity = (topConsumerStatsResult: PrometheusResponse['data']['result']) => {
+  const resourceValues = _.flatMap(topConsumerStatsResult, (resource) => resource.values);
+  const maxCapacity = _.maxBy(resourceValues, (value) => Number(value[1]));
+  return humanizeBinaryBytesWithoutB(Number(maxCapacity[1]));
```
Comment (rawagner) [inline frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/top-consumers-card/top-consumers-card-body.tsx]: no need to convert to number as `humanize` functions can handle strings too

**Pull request merged**